ibis/expr/operations.py

```diff
@@ -1310,18 +1336,22 @@ class SimpleCase(ValueOp):
 
     def __init__(self, base, cases, results, default):
         assert len(cases) == len(results)
-        ValueOp.__init__(self, base, cases, results, default)
+        super(SimpleCase, self).__init__(base, cases, results, default)
 
     def root_tables(self):
         base, cases, results, default = self.args
-        all_exprs = [base] + cases + results
-        if default is not None:
-            all_exprs.append(default)
+        all_exprs = [base] + cases + results + list(filter(
+            lambda expr: expr is not None,
+            [default]
+        ))
```

Comment:
That's a lot of fu for conditionally appending an element to a list =) but all one expression

Reply:
Yeah this needs to be cleaned up :)